### ISSUE TITLE
Regression SC2183 when printf is given '*' arguments in the format string

### ISSUE THREAD

**Issue by bendoerr** [koalaman/shellcheck#1310]
#### For bugs
- Rule Id (if any, e.g. SC1000): **SC2183**
- My shellcheck version (`shellcheck --version` or "online"): **0.5.0**
- [**yes**] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [**n/a**] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
printf "%-*s  %s\\n" "$max_arg" "$first" "$line"

```

#### Here's what shellcheck currently says:

```
In functions/common_logging.sh line 166:
      printf "%-*s  %s\\n" "$max_arg" "$first" "$line"
             ^-- SC2183: This format string has 2 variables, but is passed 3 arguments.
```

#### Here's what I wanted or expected to see:

I expect there to be no error as this is valid. This is a regression of #1184 , also version 0.4.7 does not flag this line as expected.

**Comment by koalaman** [comment]
Kindly fixed by @ngzhian. Thanks!

**Comment by bendoerr** [comment]
Thanks!
